integration/vite-route-config-test.ts

```diff
@@ -0,0 +1,348 @@
+import fs from "node:fs/promises";
+import path from "node:path";
+import { expect, type BrowserContext, type Page } from "@playwright/test";
+
+import {
+  type Files,
+  createProject,
+  viteBuild,
+  test,
+  viteConfig,
+  createEditor,
+} from "./helpers/vite.js";
+
+const js = String.raw;
+
+// This is a workaround for caching issues in WebKit
+async function reloadPage({
+  browserName,
+  page,
+  context,
+}: {
+  browserName: string;
+  page: Page;
+  context: BrowserContext;
+}): Promise<Page> {
+  if (browserName === "webkit") {
+    let newPage = await context.newPage();
+    let url = page.url();
+    await page.close();
+    await newPage.goto(url, { waitUntil: "networkidle" });
+    return newPage;
+  }
+
+  await page.reload();
+  return page;
+}
+
+test.describe("route config", () => {
+  test("fails the build if routes option is used", async () => {
+    let cwd = await createProject({
+      "vite.config.js": `
+        import { vitePlugin as remix } from "@remix-run/dev";
+
+        export default {
+          plugins: [remix({
+            routes: () => {},
+          })]
+        }
+      `,
+      "app/routes.ts": `export default INVALID(`,
```

Comment:
I don't think this is intended to be an invalid syntax file - maybe invalid copy/paste from line 86?  I think this test probably wants to assert that even with a valid `routes.ts` we error if you're using the `routes` config?

Reply:
Ah, nice catch 👍